scipy/linalg/_basic.py

```diff
@@ -317,23 +327,42 @@ def solve(a, b, lower=False, overwrite_a=False,
     return x
 
 
-def _lange_diagonal(_, a):
+def _matrix_norm_diagonal(_, a, check_finite):
     # Equivalent of dlange for diagonal matrix, assuming
     # norm is either 'I' or '1' (really just not the Frobenius norm)
-    return np.abs(np.diag(a)).max()
+    d = np.diag(a)
+    d = np.asarray_chkfinite(d) if check_finite else d
+    return np.abs(d).max()
 
 
-def _lange_tridiagonal(norm, a):
+def _matrix_norm_tridiagonal(norm, a, check_finite):
     # Equivalent of dlange for tridiagonal matrix, assuming
     # norm is either 'I' or '1'
     if norm == 'I':
         a = a.T
-    d = np.abs(np.diag(a))
-    d[1:] += np.abs(np.diag(a, 1))
-    d[:-1] += np.abs(np.diag(a, -1))
+    # Context to avoid warning before error in cases like -inf + inf
+    with np.errstate(invalid='ignore'):
```

Comment:
Probably better to make these noisy? It should probably be caught beforehand in the solve checks I reckon. Since we can't push it to lapack infs and nans anyways.

Reply:
Isn't that that job of `asarray_chkfinite` that is called right after this? 

- If `check_finite is True`, we wouldn't want to emit a warning and then raise an error for the same thing. So I'm pretty sure we want to silence the warning in that case.
- If `not check_finite` I can see it either way. We can go through the extra complexity of warning in one case but not the other if you want, but I think it is also reasonable not to warn. After all, if the user explicitly opted out of finiteness checks, they either know that all their data is finite or they accept the consequences.

Reply:
I misunderstood what this function was doing so no worries.